packages/schema-derive/src/msg.rs

```diff
@@ -0,0 +1,107 @@
+use syn::{parse_quote, DeriveInput};
+
+pub fn msg_impl(input: DeriveInput) -> DeriveInput {
+    match input.data {
+        syn::Data::Struct(_) => parse_quote! {
+            #[derive(
+                serde::Serialize,
+                serde::Deserialize,
+                Clone,
+                Debug,
+                PartialEq,
```

Comment:
While we are here, can we derive `Eq` instead of `PartialEq` as well as `Hash`?

Reply:
If we derive `Eq`, we won't be able to include any types that are not `Eq` in struct fields. Is that what we want?

```rust
26  | #[cw_serde]
    | ----------- in this procedural macro expansion
...
30  |         amount: Vec<Coin>,
    |         ^^^^^^^^^^^^^^^^^ the trait `Eq` is not implemented for `Coin`
```

Reply:
Hmm, the only class of types I am aware of that is not `Eq` is floats because NaN != NaN.

Having/requiring `Eq` would allow the use of messages as HashSet elements and HashMap keys. But I guess nobody needs that. Floats on the other hand might enter the stage eventually.

I think we can keep it as is.